### PR TITLE
[DS-6091][standalone] Fix DolphinPluginDiscovery load plugin fails on Windows os

### LINKED ISSUE #6091: [Bug][dolphinscheduler-standalone-server] StandaloneServer run failed on Windows system
Since the directory separator is not the same for different operating systems, reading the pom.xml file of registry in Windows system fails.
The error message is as follows :
```
Exception in thread "main" java.nio.file.InvalidPathException: Illegal char <:> at index 2: ../dolphinscheduler/dolphinscheduler-standalone-server/target/classes/\../../../dolphinscheduler-registry-plugin/dolphinscheduler-registry-zookeeper/pom.xml
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:229)
	at java.base/java.nio.file.Path.of(Path.java:147)
	at java.base/java.nio.file.Paths.get(Paths.java:69)
	at org.apache.dolphinscheduler.server.StandaloneServer.main(StandaloneServer.java:75)
```

### PR DESCRIPTION
**Pull request by LiuBodong**
## Purpose of the pull request
Fix the bug that StandaloneServer run failed on Windows system.

## Brief change log

* Fix bug when use "/target/classes" to judge path will always return false. 

The bug's reason is use `File#getPath.endsWith("/target/classes")` to judge if the given path is compiled class path. On Windows, the delimiter of path is '\', so the method always returns false.

The solution is use `Paths.endsWith(Path other)` method to judge, it will work correctly.

resolves #6091

### CODE REVIEW

File: dolphinscheduler-standalone-server/src/main/java/org/apache/dolphinscheduler/server/StandaloneServer.java [hunk 1]
```diff
@@ -71,10 +75,11 @@ public static void main(String[] args) throws Exception {
     }
 
     private static void startAlertServer() {
-        final Path alertPluginPath = Paths.get(
-                StandaloneServer.class.getProtectionDomain().getCodeSource().getLocation().getPath(),
-                "../../../dolphinscheduler-alert-plugin/dolphinscheduler-alert-email/pom.xml"
-        ).toAbsolutePath();
+
```
Comment (kezhenxu94) [inline dolphinscheduler-standalone-server/src/main/java/org/apache/dolphinscheduler/server/StandaloneServer.java]:
```suggestion
```

File: dolphinscheduler-standalone-server/src/main/java/org/apache/dolphinscheduler/server/StandaloneServer.java [hunk 2]
```diff
@@ -118,13 +123,54 @@ private static void startDatabase() throws IOException, SQLException {
     }
 
     private static void setTaskPlugin() {
-        final Path taskPluginPath = Paths.get(
-                StandaloneServer.class.getProtectionDomain().getCodeSource().getLocation().getPath(),
-                "../../../dolphinscheduler-task-plugin/dolphinscheduler-task-shell/pom.xml"
-        ).toAbsolutePath();
+
```
Comment (kezhenxu94) [inline dolphinscheduler-standalone-server/src/main/java/org/apache/dolphinscheduler/server/StandaloneServer.java]:
```suggestion
```

**Comment by kezhenxu94** [comment]
This PR is duplicate with #6101, can you please coordinate with @QuakeWang which PR we should continue?

**Comment by LiuBodong** [comment]
> 
> 
> This PR is duplicate with #6101, can you please coordinate with @QuakeWang which PR we should continue?

I already send an e-mail to @QuakeWang , we will talk later

**Comment by QuakeWang** [comment]
We have discussed and come up with the following recommendation: This pr retains the changes to DolphinPluginDiscovery. The changes to StandaloneServer are still based on #6101.

**Comment by LiuBodong** [comment]
I just revert the modification of StandaloneServer.java, now the pr only contains modification of DolphinPluginDiscovery.java. Please do a code review~

**Comment by kezhenxu94** [comment]
@LiuBodong please update the PR title and commit message accordingly

**Comment by LiuBodong** [comment]
already fixed in another pr

**Pull request closed without merging**